### DISCUSSION TITLE
Not being able to use Week days in the time module

### DISCUSSION THREAD

**Discussion by bwubta21122** [vlang/v#19673] [General]
### Describe the bug

Code: https://vosca.dev/p/ad6f80ef66

```v
import time

fn main() {
	c := 'Fri Oct 27 2023'
	s := time.parse_format(c, 'DDD MMM DD YYYY') or { panic(err) }
	a := s.str()
	t := time.parse(a) or { panic('failing format: ${a} | err: ${err}') }
	println(t)
}

```

### Reproduction Steps

Just run the above code provided. Thank you for your help.

### Expected Behavior

2023-10-27 00:00:00

### Current Behavior

Output:

```
V panic: time.TimeParseError: Invalid time format code: 0, error: invalid string: "Fri"!="DDD" at: 3
v hash: 0b0499b
/tmp/v_60000/../../../../../../home/admin/v/vlib/builtin/builtin.c.v:119: at _v_panic: Backtrace
/tmp/v_60000/../../../../../../box/code.v:5: by main__main
/tmp/v_60000/../../../../../../tmp/v_60000/code.7602164284236378996.tmp.c:17023: by main
Exited with error status 1
```

### V version

V 0.3.3 e738d67

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.1 3a91a5e.0b0499b
OS: linux, Debian GNU/Linux 11 (bullseye) (VM)
Processor: 1 cpus, 64bit, little endian, Intel(R) Xeon(R) CPU E5-2686 v4 @ 2.30GHz

getwd: /home/admin/playground
vexe: /home/admin/v/v
vexe mtime: 2023-09-16 11:21:36

vroot: OK, value: /home/admin/v
VMODULES: OK, value: /.vmodules
VTMP: OK, value: /tmp/v_0

Git version: git version 2.30.2
Git vroot status: Error: fatal: detected dubious ownership in repository at '/home/admin/v'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v
.git/config present: true

CC version: cc (Debian 10.2.1-6) 10.2.1 20210110
thirdparty/tcc status: Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc
 Error: fatal: detected dubious ownership in repository at '/home/admin/v/thirdparty/tcc'
To add an exception for this directory, call:

	git config --global --add safe.directory /home/admin/v/thirdparty/tcc

```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by ttytm** [comment]
It's not indicated that `DDD` is supported to parse week days by name. So I would classify this rather as feature request than as a bug.

**Reply by bwubta21122** [reply]
So how do I solve this problem?

**Comment by JalonSolov** [comment]
It also opens a whole can o' worms... should it support only English day names?  Or have full i18n support for all languages?

**Reply by bwubta21122** [reply]
So how do I solve this problem?

**Reply by JalonSolov** [reply]
Right now, you don't.  As stated, V doesn't currently support using either full or partial day names.  This discussion is open to decide _how_ to add this support.

**Reply by ttytm** [reply]
> So how do I solve this problem? @bwubta21122 

Can you state more exactly what real problem should be solved?

If you just want to get the time parsed and work with it, why not cut off the week day names, and parse the string from there? This will not work for all cases but for some - depending on what should be achieved.

**Reply by JalonSolov** [reply]
Right.  The only workaround at present it to not use day names at all.  This works as expected:
```v
import time

fn main() {
        c := 'Oct 27 2023'
        s := time.parse_format(c, 'MMM DD YYYY') or { panic(err) }
        a := s.str()
        t := time.parse(a) or { panic('failing format: ${a} | err: ${err}') }
        println(t)
}
```
with output of
```
2023-10-27 00:00:00
```

**Reply by bwubta21122** [reply]
No this is for a project where `c := 'Fri Oct 27 2023'` is the format of data entry.

**Reply by ttytm** [reply]
You could still programmatically cut off week days from the input data string and use them in the program.

Could you sketch an example result of what you are trying to achieve? This would allow to help you better as this might enable a number approaches to solve an actual problem, instead of having us fiddle around if things would work in one theoretical way.
